File: code/richediting/src/main/java/org/fbme/ide/richediting/adapters/fbnetwork/PortBase.java
package org.fbme.ide.richediting.adapters.fbnetwork;

import org.fbme.lib.iec61499.descriptors.FBPortDescriptor;
import org.fbme.lib.iec61499.fbnetwork.EntryKind;

public class PortBase implements Port {
    private final EntryKind connectionKind;

    public PortBase(FBPortDescriptor port) {
        this.connectionKind = port.getConnectionKind();
    }

    public EntryKind getConnectionKind() {
        return connectionKind;
    }
}

File: code/richediting/src/main/kotlin/org/fbme/ide/richediting/adapters/fbnetwork/PortBase.kt
package org.fbme.ide.richediting.adapters.fbnetwork

import org.fbme.lib.iec61499.descriptors.FBPortDescriptor
import org.fbme.lib.iec61499.fbnetwork.EntryKind

open class PortBase(port: FBPortDescriptor) : Port {
    override val connectionKind: EntryKind = port.connectionKind
}
